dotCMS/src/main/java/com/dotcms/security/apps/AppsAPIImpl.java

```diff
@@ -68,24 +78,59 @@
             .setVisibility(PropertyAccessor.FIELD, Visibility.ANY)
             .findAndRegisterModules();
 
-    private String toJsonString(final AppSecrets object) throws DotDataException {
+    char[] toJsonAsChars(final AppSecrets object) throws DotDataException {
         try {
-            return jsonMapper.writeValueAsString(object);
+            final byte [] bytes = jsonMapper.writeValueAsBytes(object);
+            return bytesToCharArrayUTF(bytes);
         } catch (IOException e) {
             throw new DotDataException(e);
         }
     }
 
-    private AppSecrets readJson(final String json) throws DotDataException {
+     AppSecrets readJson(final char[] chars) throws DotDataException {
         try {
-            return jsonMapper.readValue(json, AppSecrets.class);
+            final byte [] bytes = charsToBytesUTF(chars);
+            return jsonMapper.readValue(bytes, AppSecrets.class);
         } catch (IOException e) {
             throw new DotDataException(e);
         }
     }
 
-    private AppSecrets readJson(final char[] chars) throws DotDataException {
-        return readJson(String.valueOf(chars));
+    private String toJsonAsString(final AppSecrets object) throws DotDataException {
+        try {
+            return jsonMapper.writeValueAsString(object);
+        } catch (IOException e) {
+            throw new DotDataException(e);
+        }
+    }
+
+    /**
+     * This method takes a byte array and converts its contents into a char array
+     * No String middle man is created.
+     * https://stackoverflow.com/questions/8881291/why-is-char-preferred-over-string-for-passwords
+     * @param bytes
+     * @return
+     */
+    char[] bytesToCharArrayUTF(byte[] bytes) throws IOException {
+        final BufferedReader reader = new BufferedReader(
+                new InputStreamReader(new ByteArrayInputStream(bytes), StandardCharsets.UTF_8));
+        final List<Integer> integers = new ArrayList<>(bytes.length);
+        int chr;
+        while ((chr = reader.read()) != -1) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid assignments in operands](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5329810)